editor/store/effects.js

```diff
@@ -102,41 +100,66 @@ export default {
 			content: getEditedPostContent( state ),
 			id: post.id,
 		};
-
-		dispatch( {
-			type: 'UPDATE_POST',
-			edits: toSend,
-			optimist: { type: BEGIN, id: POST_UPDATE_TRANSACTION_ID },
-		} );
-		dispatch( removeNotice( SAVE_POST_NOTICE_ID ) );
 		const basePath = wp.api.getPostTypeRoute( getCurrentPostType( state ) );
-		wp.apiRequest( { path: `/wp/v2/${ basePath }/${ post.id }`, method: 'PUT', data: toSend } ).then(
-			( newPost ) => {
-				dispatch( resetPost( newPost ) );
-				dispatch( {
-					type: 'REQUEST_POST_UPDATE_SUCCESS',
-					previousPost: post,
-					post: newPost,
-					edits: toSend,
-					optimist: { type: COMMIT, id: POST_UPDATE_TRANSACTION_ID },
-				} );
-			},
-			( err ) => {
-				dispatch( {
-					type: 'REQUEST_POST_UPDATE_FAILURE',
-					error: get( err, [ 'responseJSON' ], {
-						code: 'unknown_error',
-						message: __( 'An unknown error occurred.' ),
-					} ),
-					post,
-					edits,
-					optimist: { type: REVERT, id: POST_UPDATE_TRANSACTION_ID },
+		const isAutosave = action.options && action.options.autosave;
+
+		if ( isAutosave ) {
+			toSend.parent = post.id;
+			wp.apiRequest( { path: `/wp/v2/${ basePath }/${ post.id }/autosaves`, method: 'POST', data: toSend } ).then(
+				( autosave ) => {
+					dispatch( {
+						type: 'RESET_AUTOSAVE',
+						post: autosave,
+					} );
+
+					dispatch( {
+						type: 'REQUEST_POST_UPDATE_SUCCESS',
+						previousPost: post,
+						post: post,
+						isAutosave: true,
+					} );
+				},
+				() => {
+					dispatch( {
+						type: 'RESET_AUTOSAVE',
```

Comment:
Is the behavior here to revert back to the previous state when an autosave fails? This is already accomplished for the proper save via `redux-optimist`, and it seems like we could leverage the same here. We might not want to do it on a common `UPDATE_POST` (or do we?), but maybe a separate dummy action which includes the `optimist` meta? See below in the `else` case for `optimist` property handling.

Reply:
The reason we dispatch `RESET_AUTOSAVE` here is to set the 'isAutosaving' state to false (which in turn is reflected in the post publish button and post-saved-state display... see https://github.com/WordPress/gutenberg/blob/07cccd74f52814733142600ca0eec2abc460fa84/editor/store/reducer.js#L592

Is redux-optimist appropriate for this use case?

Reply:
As I recall, Redux optimist behaves in a way that an action dispatched can either be committed (confirmed) or reverted, in which case the original dispatched action is pretended to have never existed, though all other subsequent will proceed as expected. It could be relevant here, though seems like it could be just as well to reset explicitly.

Reply:
I think its fine to set this explicitly here. Optimist is really useful when you want to show the ui state completed (optimistically) then revert that if it somehow fails. In this case it seems fine to show the autosave is working until it completes or fails.